File: main/src/main/java/com/sedmelluq/discord/lavaplayer/source/youtube/YoutubeTrackDetails.java
package com.sedmelluq.discord.lavaplayer.source.youtube;

import com.sedmelluq.discord.lavaplayer.tools.io.HttpInterface;
import com.sedmelluq.discord.lavaplayer.track.AudioTrackInfo;

import java.util.List;

public interface YoutubeTrackDetails {
    AudioTrackInfo getTrackInfo();

    List<YoutubeTrackFormat> getFormats(HttpInterface httpInterface, YoutubeSignatureResolver signatureResolver);

    String getPlayerScript();
}

File: main/src/main/java/com/sedmelluq/discord/lavaplayer/source/youtube/YoutubeTrackDetails.kt
package com.sedmelluq.discord.lavaplayer.source.youtube

import com.sedmelluq.discord.lavaplayer.tools.io.HttpInterface
import com.sedmelluq.discord.lavaplayer.track.AudioTrackInfo

interface YoutubeTrackDetails {
    val trackInfo: AudioTrackInfo
    val playerScript: String?

    fun getFormats(httpInterface: HttpInterface, signatureResolver: YoutubeSignatureResolver): List<YoutubeTrackFormat>
}
